.github/workflows/cleanup_issues_before_2021.yml

```diff
@@ -0,0 +1,60 @@
+name: Close Pre-2021 Issues
+on:
+  # schedule:
+    # - cron: '35 15 * * *'  # Runs daily at 3:30 PM - stop after first run
+  workflow_dispatch: # Allows manual triggering from the Github UI
+
+permissions:
+  issues: write  # Required to close issues and post comments
+  contents: read  # Needed by most GitHub Actions
+  pull-requests: read  # Needed to access PR data
+
+jobs:
+  close-inactive:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          
+      - name: Label old issues
+        run: |
+          # Set the cutoff date (issues created before this date will be labeled)
+          CUTOFF_DATE="2021-01-01"
+          LABEL="stale-before-2021"
+          
+          echo "Labeling issues created before $CUTOFF_DATE with $LABEL"
```

Comment:
Purely subjective opinion here but I would break up the script runner for fetching labels vs closing them (you can hand data off between action steps as variables: https://www.edwardthomson.com/blog/github_actions_15_sharing_data_between_steps)

Reply:
(this would let us use the issue helper I mentioned above as we could use a script to fetch exactly what we want but then use the helper to do other stuff like set labels/close/lock/ect)

Reply:
Yes, you're absolutely right. As I said in my previous comment, this is an exploration. I'm only acting on a tiny subset to make sure the logic is sound and I'm not messing up the repo. I agree that this is not well named. I will edit so this is inly labelling. The closing part will come later today once I confirm that this is working well. Then i'll edit to make it work for all issues prior to 2024 like the RFC stated.